dotnet/Vaas/src/Vaas/Vaas.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.WebSockets;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Vaas.Messages;
+using Websocket.Client;
+using Websocket.Client.Exceptions;
+
+namespace Vaas
+{
+    public class Vaas
+    {
+        private string Token { get; }
+
+        private WebsocketClient? Client { get; set; }
+
+        private string? SessionId { get; set; }
+
+        private bool Authenticated { get; set; }
+        
+        public Uri Url { get; set; } = new("wss://gateway-vaas.gdatasecurity.de");
+
+        private Dictionary<string, VerdictResponse> VerdictResponsesDict { get; } = new();
+
+        public Vaas(string token)
+        {
+            Token = token;
+        }
+
+        public void Connect()
+        {
+            Client = new WebsocketClient(Url, CreateWebsocketClient());
+            Client.ReconnectTimeout = null;
+            Client.MessageReceived.Subscribe(msg =>
+            {
+                if (msg.MessageType != WebSocketMessageType.Text) return;
+                var message = JsonSerializer.Deserialize<Message>(msg.Text);
+                switch (message?.Kind)
+                {
+                    case "AuthResponse":
+                        var authenticationResponse = JsonSerializer.Deserialize<AuthenticationResponse>(msg.Text);
+                        if (authenticationResponse is {Success: true})
+                        {
+                            Authenticated = true;
+                            SessionId = authenticationResponse.SessionId;
+                        }
+                        break;
+                        
+                    case "VerdictResponse":
+                        var options = new JsonSerializerOptions {Converters = {new JsonStringEnumConverter()}};
+                        var verdictResponse = JsonSerializer.Deserialize<VerdictResponse>(msg.Text,options);
+                        VerdictResponsesDict.Add(verdictResponse?.Guid ?? throw new InvalidOperationException(), verdictResponse);
+                        break;
+                }
+            });
+            Client.Start().GetAwaiter().GetResult();
+            if (!Client.IsStarted)
+            {
+                throw new WebsocketException("Could not start client");
+            }
+            
+            Authenticate();
+            Task.Run(SendPing);

```

Comment:
Remove. Websocket have their own keepalive mechanism (ping, pong, frames). The library takes care of that.

Reply:
Lib we use here unfortunatly not. It only reacts on a ping it gets but does not send any by itself. IMHO we implemented it wrong on the server side :(

Reply:
Another possible reason could have been a Thread.Sleep(40s), which prevents all async code from running in the same thread. We encountered this with Python.

Added a test, works.